frontend/src/metabase/components/FormCategoryInput/CategoryFieldPicker.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+
+import CategoryFieldInput from "./CategoryFieldInput";
+import { CategoryWidgetProps } from "./types";
+
+import CategoryRadioPicker from "./CategoryRadioPicker";
+
+const MAX_DISTINCT_OPTIONS_FOR_RADIO_INPUT = 7;
+
+function CategoryFieldPicker({
+  value,
+  onChange,
+  fieldInstance,
+}: CategoryWidgetProps) {
+  const distinctCount = fieldInstance.fingerprint?.global?.["distinct-count"];
```

Comment:
nit: do you know that `fingerprint` is completely unreliable since they only scan first 2000 rows?

Reply:
That's unfortunate but good to know. I think it should work quite well for this component's purposes though